packages/integrations/image/src/vite-plugin-astro-image.ts

```diff
@@ -39,8 +49,8 @@ export function createPlugin(config: AstroConfig, options: Required<IntegrationO
 			}
 		},
 		async load(id) {
-			// only claim image ESM imports
-			if (!filter(id)) {
+			// only claim image ESM imports and ignore vue's absolute imports
+			if (!filter(id) || absoluteImageImports.has(id)) {
```

Comment:
I don't think we can change how we load files based on `resolveId` and the `source`/`importer`. This can cause a race condition where if we're trying to load an image in both an Astro file and Vue file.

For example, if we encounter the Vue file first, the image import would returning a URL as default export. Then, when we encounter the Astro file, it wouldn't transform the image as an object anymore, it uses the cached URL default export.

Unfortunately I don't have an idea of how to fix this yet. Perhaps @Princesseuh's work on image integration next could fix it.

A workaround could be to do https://github.com/withastro/astro/issues/5924#issuecomment-1410744335